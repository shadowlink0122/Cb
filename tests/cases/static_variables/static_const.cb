// static const変数のテスト
int process(){
    static const int MAX_VALUE = 100;
    static int counter = 0;
    
    if(counter + 15 <= MAX_VALUE){
        counter += 15;
    }
    
    return counter;
}

int main(){
    // static const値は変更できないが、static変数は変更可能
    for(int i = 0; i < 8; i++){
        println(process());
    }
    return 0;
}
