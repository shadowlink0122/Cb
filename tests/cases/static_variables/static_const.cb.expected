15
30
45
60
75
90
90
90
